src/displayapp/screens/WatchFaceInfineat.cpp

```diff
@@ -434,12 +434,13 @@ void WatchFaceInfineat::Refresh() {
 
   batteryPercentRemaining = batteryController.PercentRemaining();
   isCharging = batteryController.IsCharging();
-  if (batteryController.IsCharging()) { // Charging battery animation
-    chargingBatteryPercent += 1;
+  if (batteryController.IsCharging() && (lv_tick_get() - savedAnimationTick > 150)) { // Charging battery animation
```

Comment:
What happens when the tick counter wrapps around. Then the tick count is smaller than the saved tick count. What would be the resulting behaviour?

Reply:
Good point. I need to add a cast to fix this ```(uint32_t(lv_tick_get() - savedAnimationTick) > 150)```

Reply:
> Good point. I need to add a cast to fix this `(uint32_t(lv_tick_get() - savedAnimationTick) > 150)` Now, wrap around shouldn't be a problem.

I tested on the simulator with and without the ```uint32_t``` and the compiler applies the same modulo arithmetic for unsigned ints to either version with no change in the complied flash and ram sizes. However, it's probably safer to leave ```uint32_t``` in.